src/main/java/eu/europeana/metis/sandbox/service/util/XsltUrlUpdateServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package eu.europeana.metis.sandbox.service.util;
+
+import eu.europeana.metis.sandbox.entity.TransformXsltEntity;
+import eu.europeana.metis.sandbox.repository.TransformXsltRepository;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.nio.charset.StandardCharsets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class XsltUrlUpdateServiceImpl implements XsltUrlUpdateService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(XsltUrlUpdateServiceImpl.class);
+  private static final HttpClient httpClient = HttpClient.newBuilder().build();
+  private final TransformXsltRepository transformXsltRepository;
+
+  public XsltUrlUpdateServiceImpl(TransformXsltRepository transformXsltRepository) {
+    this.transformXsltRepository = transformXsltRepository;
+  }
+
+  @Override
+  @Transactional
+  public void updateXslt(String defaultXsltUrl) {
+
+    HttpRequest httpRequest = HttpRequest.newBuilder()
+        .GET()
+        .uri(URI.create(defaultXsltUrl))
+        .build();
+
+    try (final InputStream xsltStream = httpClient.send(httpRequest, BodyHandlers.ofInputStream())
+        .body()) {
+      String transformXslt = new String(xsltStream.readAllBytes(),
+          StandardCharsets.UTF_8);
+      var entity = transformXsltRepository.findByTransformXslt(transformXslt);
+
+      if (entity != null && entity.isPresent()) {
```

Comment:
If you have it in the database, do you need to save it again?

Reply:
Also not sure if the id field in the TransformXsltEntity is useful since you are using the uniqueness here by the transformXslt String content.
Which means the xslt entity can grow indefinitely?
I think what we should better do is re-use the returned object of the database by checking if there is one and only just one(without checking it's content on the query), and then we can update the specific entity(now comparing the content with what we have from the http and decide if we need to update that specific entity) with the same id and not create new ones every time.